include/service/entities/site_service.js

```diff
@@ -554,5 +569,54 @@ module.exports = function SiteServiceModule(pb) {
         return url.format(urlObject).replace(/\/$/, '');
     };
 
+    SiteService.prototype.deleteSiteSpecificContent = function (collections, siteid, callback) {
+        var self = this;
+
+        SiteService.siteExists(siteid, function (err, exists) {
+            if (util.isError(err)) {
+                return callback(err);
+            }
+
+            if (!exists) {
+                return callback(pb.BaseObjectService.validationError([pb.BaseObjectService.validationFailure("siteid", "Invalid siteid")]));
+            }
+
+            var tasks = util.getTasks(collections, function (collections, i) {
+                return function (taskCallback) {
+                    self.deleteSingle({where: {site: siteid}, collection: collections[i].name}, function (err, commandResult) {
```

Comment:
Using delete single here doesn't make sense.  Using a specific object service type here for other object types can be dangerous.  It will emit events for objects that it doesn't apply to.  In this particular case just use the DAO prototype so that you can manually specify the collections and not alter the behavior of the BaseObjectService.  

The long term solution will be to create a factory such that BaseObjectService instances for a specific type can be looked up through a single function.  Until then it will be fine to delete the site specific data for each collection through DAO.